flask_dance/contrib/reddit.py

```diff
@@ -0,0 +1,107 @@
+from __future__ import unicode_literals
+
+from flask_dance.consumer import OAuth2ConsumerBlueprint, OAuth2Session
+from functools import partial
+from flask.globals import LocalProxy, _lookup_app_object
+
+from flask_dance import __version__ as _flask_dance_version
+
+try:
+    from flask import _app_ctx_stack as stack
+except ImportError:
+    from flask import _request_ctx_stack as stack
+
+__maintainer__ = "Sergey Storchay <r8@r8.com.ua>"
+
+
+def _default_user_agent():
+    return "Flask-Dance/%s" % _flask_dance_version
+
+
+class RedditOAuth2Session(OAuth2Session):
+    def __init__(self, *args, **kwargs):
+        super(RedditOAuth2Session, self).__init__(*args, **kwargs)
+
+        # Reddit requires to set non-generic user agent
+        self.headers["User-Agent"] = self.blueprint.user_agent or _default_user_agent()
+
+    def fetch_token(self, *args, **kwargs):
+        # Pass client_id to session so it could trigger Basic Auth
+        return super(RedditOAuth2Session, self).fetch_token(
+            client_id=self.blueprint.client_id, *args, **kwargs)
+
+
+def make_reddit_blueprint(user_agent=None,
```

Comment:
Please put the `user_agent` argument at the *end* of the list of arguments, not at the start. Every other configuration has `client_id` and `client_secret` as the first two arguments, so that the developer can pass them as positional arguments. The Reddit configuration should match that convention.